pkg/clients/acm/certificate.go

```diff
@@ -74,6 +74,21 @@ func GenerateCreateCertificateInput(name string, p *v1alpha1.CertificateParamete
 
 // GenerateCertificateStatus is used to produce CertificateExternalStatus from acm.certificateStatus
 func GenerateCertificateStatus(certificate acm.CertificateDetail) v1alpha1.CertificateExternalStatus {
+
+	if certificate.Type == acm.CertificateTypeAmazonIssued && certificate.DomainValidationOptions[0].ResourceRecord != nil {
```

Comment:
Did you consider adding another condition here to ensure that the whole provider will not panic if `DomainValidationOptions` is not present or zero length? normally I would check `len(certificate.DomainValidationOptions) > 0` before accessing `certificate.DomainValidationOptions[0]`

Reply:
@chlunde  thanks for review - i will check it